### ISSUE TITLE
Releasing mouse button outside calendar while dragging cancels adding an event

### ISSUE THREAD

**Issue by jeanlazarou** [jquense/react-big-calendar#1636]
#### Do you want to request a _feature_ or report a _bug_?

Report a _bug_, because previous version did not present the issue.

#### What's the current behavior?

v0.23 (also 0.24)

When you drag on the calendar to add en event, if you release the mouse button (drag end) outside the calendar area, the dragging is cancelled.

#### What's the expected behavior?

I would expect that ending the drag anywhere outside the calendar is like releasing the mouse button at the bottom of the calendar (if dragging that direction)

I made a workaround in a fork of mine that is **totally wrong**.

The change I made is in `Selection.js`, in the function `_handleTerminatingEvent`:

```diff
-    let inRoot = !this.container || contains(this.container(), e.target)
+    let inRoot = true
```

As you can see it totally denies something that appears important.

**Comment by jhnns** [comment]
I agree, imo this is a bug. It has been introduced a while ago with https://github.com/jquense/react-big-calendar/pull/1119

In order to preserve the intended fix from #1119, I think the patch in `Selection.js` should look like this:

```diff
diff --git a/node_modules/react-big-calendar/lib/Selection.js b/node_modules/react-big-calendar/lib/Selection.js
index e5cb25b..149b445 100644
--- a/node_modules/react-big-calendar/lib/Selection.js
+++ b/node_modules/react-big-calendar/lib/Selection.js
@@ -333,15 +333,15 @@ function () {
       return this.emit('reset');
     }
 
+    if (click) {
+      return this._handleClickEvent(e);
+    }
+
     if (!inRoot) {
       return this.emit('reset');
     }
     
-    if (click && inRoot) {
-      return this._handleClickEvent(e);
-    } // User drag-clicked in the Selectable area
-
-
+    // User drag-clicked in the Selectable area
     if (!click) return this.emit('select', bounds);
   };
```

*⚠️ However, I don't know if this patch would break drag'n'drop*
